sw/host/opentitantool/src/command/rom_patch.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use log::*;
+use serde_annotate::Annotate;
+use std::any::Any;
+use std::io::Read;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+use opentitanlib::app::command::CommandDispatch;
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::crypto::rsa::RsaPrivateKey;
+use opentitanlib::rom_patch::patch::RomPatchPartition;
+
+/// Generate an OTP ROM patch partition
+#[derive(Debug, StructOpt)]
+pub struct Generate {
+    #[structopt(
+        short = "p",
+        long = "patch",
+        name = "HJSON_FILE",
+        help = "HJSON formatted ROM patch description file"
+    )]
+    patch: PathBuf,
+
+    #[structopt(
+        short = "k", long = "key",
+        name = "DER_FILE",
+        parse(try_from_str=RsaPrivateKey::from_pkcs8_der_file),
+        help = "RSA private key file in PKCS#1 DER format"
+    )]
+    key: RsaPrivateKey,
+
+    #[structopt(
+        short = "s",
+        long = "otp_size",
+        name = "OTP_SIZE",
+        help = "OTP ROM patch partition total size (bytes)"
+    )]
+    otp_size: usize,
+}
+
+impl CommandDispatch for Generate {
+    fn run(
+        &self,
+        _context: &dyn Any,
+        _transport: &TransportWrapper,
+    ) -> Result<Option<Box<dyn Annotate>>> {
+        let mut patch_partition = RomPatchPartition::new(&self.patch, &self.key)?;
+        let mut buffer = vec![0_u8; self.otp_size];
+        _ = patch_partition.read(&mut buffer)?;
+
+        println!("uint8_t fake_patch[] =");
```

Comment:
We have other similar tooling in the repo to generate code from a template + hjson file using the python library [mako templates](https://www.makotemplates.org/).
Is there a reason for the tool to be inside opentitantool or can this also be a python script using a template file?
In case it needs to be inside opentitantool, can we use a template crate?

Reply:
We are now generating an HJSON OTP partition image, to be used as an OTP image overlay.